tools/kwiver_tool_runner.cxx

```diff
@@ -0,0 +1,255 @@
+    free (m_argv);
+  }
+
+  // tool runner arguments.
+  std::string m_output_file; // empty is no file specified
+
+  std::vector<std::string> m_applet_args;
+  std::string m_applet_name;
+
+  int m_argc;
+  const char** m_argv;
+
+private:
+  int m_state;
+};
+
+
+// ----------------------------------------------------------------------------
+void
+std_stream_dtor(void* /*ptr*/)
+{
+  // We don't want to delete std::cin or std::cout.
+}
+
+
+// ----------------------------------------------------------------------------
+void tool_runner_usage( applet_context_t ctxt,
+                        kwiver::vital::plugin_manager& vpm )
+{
+  // display help message
+  std::cout << "Usage: kwiver  <applet>  [args]" << std::endl
+            << "<applet> can be one of the following:" << std::endl
+            << "help - prints this message" << std::endl
+    ;
+
+  // Get list of factories for implementations of the applet
+  const auto fact_list = vpm.get_factories( typeid( kwiver::tools::kwiver_applet ).name() );
+
+  // Loop over all factories in the list and display name and description
+  for( auto fact : fact_list )
+  {
+    std::string buf = "-- Not Set --";
+    fact->get_attribute( kwiver::vital::plugin_factory::PLUGIN_NAME, buf );
+    std::cout << "    " << buf << " - ";
+
+    fact->get_attribute( kwiver::vital::plugin_factory::PLUGIN_DESCRIPTION, buf );
+    std::cout << ctxt->m_wtb.wrap_text( buf );
+  }
+}
+
+
+// ============================================================================
+int main(int argc, char *argv[])
+{
+  //
+  // Global shared context
+  // Allocated on the stack so it will automatically clean up
+  //
+  applet_context_t tool_context = std::make_shared< kwiver::tools::applet_context >();
+
+  kwiver::vital::plugin_manager& vpm = kwiver::vital::plugin_manager::instance();
+  vpm.load_all_plugins();
+
+
+  // initialize the global context
+  tool_context->m_ostream.reset( &std::cout, &std_stream_dtor );
+  tool_context->m_wtb.set_indent_string( "      " );
+
+  command_line_parser options( argc, &argv );
+
+  if ( options.m_applet_name.empty() || options.m_applet_name == "help" )
+  {
+    tool_runner_usage( tool_context, vpm );
+    return 0;
+  } // end help code
+
+  // Process tool runner options
+  if ( ! options.m_output_file.empty() )
+  {
+    tool_context->m_ostream.reset( new std::ofstream( options.m_output_file ) );
+    if (! tool_context->m_ostream->good())
+    {
+      std::cerr <<  "Unable to open output file: " << options.m_output_file << std::endl;
+      return -1;
+    }
+  }
+
+  // ----------------------------------------------------------------------------
+  try
+  {
+    // Create applet based on the name provided
+    applet_factory app_fact;
+    auto applet = app_fact.create( options.m_applet_name );
+    tool_context->m_applet_name = options.m_applet_name;
+    applet->initialize( tool_context.get() );
+
+    // check for <applet> help
+    if ( options.m_applet_args[1] == "help" )
```

Comment:
I don't think we want to hijack the applet arguments by looking into its argument list for `help`.  The syntax should be  `kwiver help <applet>` not `kwiver <applet> help`.  This is how Git does it too.  Basically `help` is like its own simple applet.  With no arguments is prints all the other applets, when passed an applet name it prints the help for that applet.   We can also support `kwiver <applet> --help`, but that will be up to each applet to handle if they want.